dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/strategy/PageViewStrategy.java

```diff
@@ -77,6 +84,40 @@ public HTMLPageAsset fromContentlet(final Contentlet contentlet) {
         map.put("statusIcons", UtilHTML.getStatusIcons(page));
         map.put("__icon__", IconType.HTMLPAGE.iconName());
 
+        final ContentletVersionInfo info = APILocator.getVersionableAPI().
+                getContentletVersionInfo(page.getIdentifier(), page.getLanguageId());
+
+        map.put("workingInode",  info.getWorkingInode());
+        map.put("shortyWorking", APILocator.getShortyAPI().shortify(info.getWorkingInode()));
+        map.put("canEdit", this.permissionAPI.doesUserHavePermission(page, PermissionLevel.EDIT.getType(), user, false));
+        map.put("canRead", this.permissionAPI.doesUserHavePermission(page, PermissionLevel.READ.getType(), user, false));
+        map.put("liveInode", info.getLiveInode());
+        map.put("shortyLive", APILocator.getShortyAPI().shortify(info.getLiveInode()));
+        map.put("canLock", canLock(page, user));
+
+        if(info.getLockedBy()!=null) {
+            map.put("lockedOn", info.getLockedOn());
+            map.put("lockedBy", info.getLockedBy());
+            map.put("lockedByName", getLockedByUserName(info));
+        }
+
         return map;
     }
+
+    private boolean canLock(final HTMLPageAsset page, User user)  {
+        try {
+            APILocator.getContentletAPI().canLock(page, user);
+            return true;
```

Comment:
toolbox already has contentletAPI. Use the one from there

Reply:
The Toolbox was generalized to be used by any class. Also it was refactored to use the Builder pattern to ease the use of it.